src/rpc/methods/state.rs

```diff
@@ -1148,3 +1157,39 @@ pub async fn state_deal_provider_collateral_bounds<DB: Blockstore + Send + Sync
         min: TokenAmount::from_atto(min),
     })
 }
+
+pub enum StateGetBeaconEntry {}
+
+impl RpcMethod<1> for StateGetBeaconEntry {
+    const NAME: &'static str = "Filecoin.StateGetBeaconEntry";
+    const PARAM_NAMES: [&'static str; 1] = ["epoch"];
+    const API_VERSION: ApiVersion = ApiVersion::V1;
+
+    type Params = (LotusJson<ChainEpoch>,);
+    type Ok = LotusJson<BeaconEntry>;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore>,
+        (LotusJson(epoch),): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        loop {
+            let genesis_timestamp = ctx.chain_store.genesis_block_header().timestamp;
+            let block_delay = ctx.state_manager.chain_config().block_delay_secs;
+            let now_epoch = crate::chain_sync::get_now_epoch(
+                chrono::Utc::now().timestamp(),
+                genesis_timestamp as _,
+                ctx.state_manager.chain_config().block_delay_secs as _,
+            );
+            if epoch <= now_epoch {
+                break;
+            }
+            tokio::time::sleep(Duration::from_secs(block_delay as _)).await;
```

Comment:
Can we deterministically calculate when we're at the right epoch? This would get rid of a loop, and allow getting the beacon almost immediately if it should be there in 1s (otherwise, we'd wait 29s for no good reason). What do you think?

Reply:
It might not be worth the extra logic. Requesting the beacon for future epochs is unlikely to happen often.

Reply:
I've made the change before seeing the discussion, plz take another look.